### PR TITLE
fix(cli): resolve catalog merge driver paths

### LINKED ISSUE #580: fix(cli): catalog merge-driver cannot infer the format from Git temp files, and --path lookup breaks when the config root is not the Git root
## 1. Without `--format`, format inference always fails in real merge-driver invocations

`crates/palamedes-cli/src/commands/catalog/merge.rs:142` passes only the explicit `--format` down; inference (`crates/palamedes/src/catalog_three_way.rs:150-158`) uses the ancestor/ours/theirs/output paths — which in a Git merge-driver invocation are all extension-less `.merge_file_XXXXXX` temp files. So every conflicted merge errors out with a confusing "cannot infer format" on temp paths. The real pathname is available via `--path %P` but is only used for the PO-options lookup. `docs/cli.md:253` claims "Inferred from paths when omitted" without the driver caveat. Note a single hardcoded `--format` is also wrong for mixed PO/FCL repos with one driver entry.

Direction: when `--path` is set and `--format` absent, infer from `--path` before falling back; document it.

## 2. `--path %P` silently loses per-catalog PO options in monorepos

`catalog_for_path` (`crates/palamedes-cli/src/config.rs:334-339`) joins a relative path onto the config root, but Git's `%P` is repo-root-relative. With a config at `apps/web/palamedes.yaml` the lookup becomes `apps/web/apps/web/locales/...` → `None` → `po: None` — so the merged catalog is re-folded with default line breaks and the next `pmds extract` turns that into spurious drift on every merge. Exactly the failure the option exists to prevent (comment at `merge.rs:130-134`).

Direction: additionally resolve `--path` against the process CWD (Git runs drivers at the worktree top) and/or match by suffix against resolved catalog paths.

*Filed from the 2026-08-07 repository audit.*

### PR DESCRIPTION
**Pull request by swernerx**
## Summary

- Make logical `--path` format inference win whenever `--format` is omitted, so one Git merge driver handles extensionless PO and FCL temporary inputs; explicit `--format` remains authoritative.
- Resolve catalog lookups through normalized exact config-root, invocation-CWD, and Git-worktree-root candidates; ambiguous paths fail rather than guessing, preserving per-catalog PO options.
- Align the README, CLI guide, and migration instructions with Git’s `%O %A %B %A --path %P` contract, backed by Git-level PO/FCL and adversarial path coverage.

## Issue

Closes #580

## Validation

- `rustup run 1.95 cargo fmt --all --check`
- `rustup run 1.95 cargo clippy --workspace --all-targets --locked -- -D warnings`
- `rustup run 1.95 cargo test --workspace --locked`
- `rustup run stable cargo fmt --all --check`
- `rustup run stable cargo clippy --workspace --all-targets --locked -- -D warnings`
- `rustup run stable cargo test --workspace --locked`
- `pnpm format:check`
- `pnpm lint`

🔧 Emily Carter · Implementation Agent

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -260,7 +260,7 @@ driver:
 
 ```bash
 git config merge.palamedes-catalog.driver \
-  'pmds catalog merge-driver %O %A %B %A --path %P --format=po --conflict-strategy=use-first'
+  'pmds catalog merge-driver %O %A %B %A --path %P --conflict-strategy=use-first'
```
Comment (swernerx) [inline README.md:263]: [P2] Update the published merge-driver examples

`packages/cli/README.md` still configures separate PO/FCL drivers with hard-coded `--format`, while `llms.txt` and `llms-full.txt` still teach the PO-only form. Those are primary npm and coding-agent entry points, so mixed-format users following them will miss the single-driver behavior this change adds. Please update all three to the `%O %A %B %A --path %P` command shown here, keeping `--format` documented only as an explicit override.

🔎 Arthur Bennett · Independent Review Agent

**Review by swernerx — COMMENTED** [review]
Verdict: changes requested

The merge behavior itself checks out: explicit `--format` stays authoritative, `--path` handles extensionless PO/FCL Git inputs, exact lookup rejects ambiguity, and the configured PO serialization path stays extract-stable.

One P2 docs gap remains in the published CLI and agent-facing surfaces; details are inline.

🔎 Arthur Bennett · Independent Review Agent

File: README.md (hunk 1 above)
Comment (swernerx) [reply README.md:263]: Addressed in a113e045f7d45127177b617d6177de9866f5ecab.

Updated `packages/cli/README.md`, `llms.txt`, and `llms-full.txt` to use the single mixed-format driver command `pmds catalog merge-driver %O %A %B %A --path %P --conflict-strategy=use-first`; each now documents `--format` only as an explicit PO/FCL override. Added an LLMS surface guard covering all three entry points and rejecting hard-coded merge-driver formats.

Focused evidence: `pnpm check:llms`, `pnpm build:site`, `pnpm format:check`, `pnpm lint`, `rustup run stable cargo test -p palamedes-cli --test catalog_merge --locked`, plus full Rust 1.95 and stable fmt/clippy/workspace-test gates all passed.

🔧 Emily Carter · Implementation Agent

File: scripts/check-llms-surface.mjs [hunk 2]
```diff
@@ -249,12 +249,25 @@ function verifyFeatureNarrative(read) {
   }
 }
 
+function verifyMergeDriverGuidance(read) {
+  const command = "pmds catalog merge-driver %O %A %B %A --path %P --conflict-strategy=use-first"
```
Comment (swernerx) [inline scripts/check-llms-surface.mjs]: [P2] Validate merge-driver arguments without fixing their spelling or order

This exact command string makes the guard reject the valid `--conflict-strategy use-first` spelling used by `docs/cli.md`, while the ordered regex below lets a stale `--format po` command through when `--format` appears before `--path` and the file retains one canonical command elsewhere. I reproduced both through injected readers against `checkLlmsSurface`. Please inspect every merge-driver command on each primary surface, reject `--format po|fcl` in either value spelling and any argument order, and accept harmless whitespace plus `--option=value`/`--option value` variants; add tests for those cases.

🔎 Arthur Bennett · Independent Review Agent

**Review by swernerx — COMMENTED** [review]
Verdict: changes requested

The three primary docs surfaces now agree, and the rebased Issue #580 implementation remains green. One P2 remains in the freshness guard: its exact-string and ordered-regex checks are over-strict for valid CLI spelling and under-strict for reordered stale `--format` commands. Details inline.

🔎 Arthur Bennett · Independent Review Agent

File: scripts/check-llms-surface.mjs (hunk 2 above)
Comment (swernerx) [reply scripts/check-llms-surface.mjs]: Addressed in 787404a36add8068461bf45493a17fb23884189e. The primary-surface guard now examines every merge-driver command in fenced or inline command examples, tokenizes both --option=value and --option value forms independent of ordering, rejects hard-coded PO/FCL formats, and requires --path %P plus %O %A %B %A. It deliberately permits both conflict-strategy spellings and ignores prose.

Coverage exercises reordered/wrapped commands, both format spellings and orders, prose, missing path, wrong placeholders, and a stale command that remains after a canonical command. pnpm test:llms (12), pnpm check:llms, pnpm format:check, pnpm lint, pnpm build:site, focused catalog_merge, and workspace Rust 1.95/stable fmt/clippy/tests passed.

🔧 Emily Carter · Implementation Agent

File: README.md (hunk 1 above)
Comment (swernerx) [reply README.md:263]: Confirmed on final fix 787404a36add8068461bf45493a17fb23884189e: packages/cli/README.md, llms.txt, and llms-full.txt remain aligned on the single mixed-format driver command with --path %P; --format is retained only as an explicit override. The strengthened per-command guard keeps that contract covered while accepting valid conflict-strategy spellings.

pnpm test:llms (12), pnpm check:llms, pnpm format:check, pnpm lint, pnpm build:site, focused catalog_merge, and workspace Rust 1.95/stable fmt/clippy/tests passed.

🔧 Emily Carter · Implementation Agent

File: scripts/check-llms-surface.mjs (hunk 2 above)
Comment (swernerx) [reply scripts/check-llms-surface.mjs]: Final rebased head: 7de61b5c7a4ddd0219cc0e680320cb19810523d5. The merge-driver guard remains the requested per-command, order-independent inspection: it rejects either hard-coded PO/FCL spelling, requires --path %P and %O %A %B %A, accepts valid conflict-strategy spellings, and ignores prose.

While validating the exact merge ref, Windows deterministically exposed a newer-main core-node fixture mismatch: its expected path normalized the configured POSIX suffix to backslashes, while the native diagnostic preserves that suffix as /. The only extra change is a two-line test expectation that preserves the native root separator and configured suffix; it is required for the Windows gate. pnpm test:llms (12), pnpm check:llms, pnpm lint, and pnpm format:check pass; the local native test cannot start because the worktree Rust 1.95 cannot compile current main's oxc dependency (cold_path), before tests execute.

🔧 Emily Carter · Implementation Agent

File: README.md (hunk 1 above)
Comment (swernerx) [reply README.md:263]: Confirmed on the rebased final head 7de61b5c7a4ddd0219cc0e680320cb19810523d5: packages/cli/README.md, llms.txt, and llms-full.txt remain aligned on the single mixed-format driver command with --path %P, and retain --format only as an explicit override. The strengthened per-command guard continues to cover that contract.

pnpm test:llms (12), pnpm check:llms, pnpm lint, and pnpm format:check pass; the exact-head seven-gate CI run is in progress.

🔧 Emily Carter · Implementation Agent

File: scripts/check-llms-surface.mjs [hunk 3]
```diff
@@ -249,12 +249,69 @@ function verifyFeatureNarrative(read) {
...
+  return [...fenced, ...inline]
+}
+
+function mergeDriverCommands(text) {
+  const commands = []
+  for (const example of codeExamples(text)) {
+    const normalized = example.replaceAll(/\\\s*\r?\n/gu, " ").replaceAll(/\s+/gu, " ")
+    const matches = [...normalized.matchAll(/\bpmds\s+catalog\s+merge-driver\b/gu)]
+    for (const [index, match] of matches.entries()) {
+      const command = normalized.slice(match.index, matches[index + 1]?.index).trim()
```
Comment (swernerx) [inline scripts/check-llms-surface.mjs]: [P2] Stop each merge-driver check at its logical command boundary

Slicing to the next merge-driver occurrence lets this command absorb every later line in the same fenced block. I reproduced an invalid driver with no placeholders being accepted because a following `printf` line contained `%O %A %B %A`; conversely, a valid driver followed by an unrelated `pmds catalog merge --format po` line is rejected. Commands still aren't inspected independently, and non-driver lines can create false results. Please split fenced examples into logical shell commands, joining only explicit backslash continuations, before tokenizing; add both boundary cases to the tests.

🔎 Arthur Bennett · Independent Review Agent

**Review by swernerx — COMMENTED** [review]
Verdict: changes requested

The rebased runtime and docs work checks out, including the Windows-only expectation repair. One P2 remains in the freshness guard: fenced examples are tokenized through the rest of the block instead of one logical shell command at a time, so adjacent lines can mask an invalid driver or falsely reject a valid one. Details inline.

🔎 Arthur Bennett · Independent Review Agent

File: scripts/check-llms-surface.mjs (hunk 3 above)
Comment (swernerx) [reply scripts/check-llms-surface.mjs]: Addressed in 5e7b206d4f2e4660744352d5ffc0ef75b632b651. Fenced examples are now split into logical shell commands before inspection: ordinary lines are independent, while only an odd trailing run of backslashes (after trailing whitespace) joins the next line. Fenced blocks are excluded from inline-span scanning, so commands cannot be duplicated or absorb later lines.

The real checkLlmsSurface suite now covers both boundary reproductions: a later printf with %O %A %B %A cannot satisfy a driver missing placeholders, and a later pmds catalog merge --format po cannot invalidate a valid driver. It also covers whitespace-tolerant continuations and escaped-backslash non-continuations, while retaining both option spellings/order, format rejection, path/placeholder requirements, and prose exclusion. pnpm test:llms/check:llms (15), format, lint, site build, focused catalog_merge (7), and Rust 1.95/stable fmt/clippy/workspace tests passed.

🔧 Emily Carter · Implementation Agent

File: scripts/check-llms-surface.mjs (hunk 2 above)
Comment (swernerx) [reply scripts/check-llms-surface.mjs]: Updated final head: 5e7b206d4f2e4660744352d5ffc0ef75b632b651. The prior per-command option contract remains intact, now with fenced shell examples split before tokenization so adjacent commands cannot mask or contaminate a merge-driver check.

Evidence: checkLlmsSurface covers order-independent --format rejection in both spellings, --path %P and %O %A %B %A requirements, both Arthur command-boundary cases, and continuation/escaped-backslash behavior; pnpm test:llms/check:llms (15), format, lint, site build, focused catalog_merge (7), and Rust 1.95/stable fmt/clippy/workspace tests passed.

🔧 Emily Carter · Implementation Agent

**Review by swernerx — COMMENTED** [review]
Verdict: clear

The logical-command splitter closes both guard findings. Fenced examples are checked one command at a time; odd trailing backslash runs join continuations while even runs stay separate, and the earlier option/order, format, path, placeholder, and prose contracts remain covered. The final commit is isolated to the guard and its tests, and the focused LLMS and merge-driver checks pass at the exact head.

🔎 Arthur Bennett · Independent Review Agent
